### ISSUE TITLE
Kernel Panic after install LCD5-show on RPi 4 in Kali Linux 2019.4 

### ISSUE THREAD

**Issue by Heggeg** [lcdwiki/LCD-show-kali#3]
- RPi 4 4 GB RAM
- Kali Linux 2019.4 fresh installed only newest updates were installed. 

Error after Reboot:
end Kernel panic - not syncing: No working init found. Try passing init= option to kernel

**Comment by soaringswine** [comment]
remove / comment out line:

`sudo echo "enable_uart=1" >> ./boot/config.txt.bak`

in the install script
or comment out "enable_uart=1" in your /boot/config.txt.

if that doesn't help, let me know as I made some other modifications to the script and got it work on Kali 2019.4 on RPi 4 but I'm pretty sure the uart line was the issue.

**Comment by Sorceree** [comment]
Hi @soaringswine I tried your solution but am getting a "E: Unable to locate package libraspberrypi-dev". Any ideas? Thanks

**Comment by soaringswine** [comment]
@Sorceree on RPi 4 with Kali 2019.4?

**Comment by Sorceree** [comment]
@soaringswine That's right yeah, I just installed Kali today so everything is default including the sourses.list. All I have done so far is connect it to my wifi and run 'apt update && apt upgrade'. Then I tried installing my LCD screen and got the same error as @Heggeg.

**Comment by soaringswine** [comment]
hmm.. to be clear, I used kali-linux-2019.4-rpi3-nexmon-64.img and I have a different LCD (MPI4008) but the scripts are pretty similar. basically, the only things that seemed to be needed to make my LCD work were:

1) /boot/config.txt entires related to HDMI modes and parameters.
2) x.org.conf.d files.

I'll post my working MPI4008-show script below, can you try and substitute your values / parameters in for the ones I have for my LCD? take note of the lines I commented out which didn't seem to matter or caused issues. it's best to git clone the entire repo and then just modify the specific script you use.

`#!/bin/bash

#sudo ./system_backup.sh
mkdir ./boot
sudo cp -rf /boot/config.txt ./boot/config.txt.bak
#sudo echo "hdmi_force_hotplug=1" >> ./boot/config.txt.bak
#sudo echo "hdmi_force_edid_audio=1" >> ./boot/config.txt.bak
sudo echo "dtparam=i2c_arm=on" >> ./boot/config.txt.bak
sudo echo "dtparam=spi=on" >> ./boot/config.txt.bak
#THIS WAS CAUSING KERNEL PANIC sudo echo "enable_uart=1" >> ./boot/config.txt.bak
sudo echo "display_rotate=3" >> ./boot/config.txt.bak
sudo echo "max_usb_current=1" >> ./boot/config.txt.bak
#sudo echo "config_hdmi_boost=7" >> ./boot/config.txt.bak
sudo echo "hdmi_group=2" >> ./boot/config.txt.bak
#sudo echo "hdmi_mode=1" >> ./boot/config.txt.bak
sudo echo "hdmi_mode=87" >> ./boot/config.txt.bak
#sudo echo "hdmi_drive=2" >> ./boot/config.txt.bak
sudo echo "hdmi_cvt=480 800 60 6 0 0 0" >> ./boot/config.txt.bak
sudo echo "dtoverlay=ads7846,cs=1,penirq=25,penirq_pull=2,speed=50000,keep_vref_on=0,swapxy=0,pmax=255,xohms=150,xmin=200,xmax=3900,ymin=200,ymax=3900" >> ./boot/config.txt.bak
sudo cp -rf ./boot/config.txt.bak /boot/config.txt

sudo cp -rf ./usr/99-fbturbo.conf-HDMI /usr/share/X11/xorg.conf.d/99-fbturbo.conf 
if [ ! -d /etc/X11/xorg.conf.d ]; then
sudo mkdir /etc/X11/xorg.conf.d 
fi
sudo cp -rf ./usr/99-calibration.conf-397-270 /etc/X11/xorg.conf.d/99-calibration.conf 
sudo touch ./.have_installed
echo "hdmi:resistance:397:270:480:800" > ./.have_installed

sudo sync
sudo sync
sleep 1
if [ $# -eq 1 ]; then
sudo ./rotate.sh $1
elif [ $# -gt 1 ]; then
echo "Too many parameters"
fi

echo "reboot now"
sudo reboot`

You could also try "apt install libraspberrypi-dev" but I still find that odd that you encountered that error and I didn't. though I believe I installed my LCD before I did an "apt upgrade" but I doubt that matters.

**Comment by n4kack** [comment]
Had the same problem on a RPi 3 and Kali Linux 2019.4 64bit

managed to get up and running by
1. Copying the /boot/config.txt file from the SD card after using balenaEtcher to flash it with the Kali image, then eject card
2. card into Pi and boot, follow the steps to clone LCD-show-kali repo 
> git clone https://github.com/lcdwiki/LCD-show-kali.git
3. chmod it and cd to it as per ReadMe (although it will be named LCD-show-kali)
4. Kernel Panic
5. Power Pi down, pull card, reconnect to computer used for flashing (where the kali config.txt file is)
6. open the config.txt file on the card and the one saved earlier
7. On the version on the card, there is only about 5 or 6 lines you need to keep, these should be the last  lines in the file, remove everything before that and paste the content of your previously saved config.txt above those few lines
It will look something like this [config.txt](https://github.com/lcdwiki/LCD-show-kali/files/4164700/config.txt)

8. save, eject, pull, insert, boot, and then probably go on the hunt for info to calibrate it.. maybe

Hope this helps - Good Luck

**Comment by bukermass** [comment]
Same Kernel Panic with Kali 2020.2b. Any solution??

With "2020-05-27-raspios-buster-full-armhf" no problems. 
Just remember to use "apt-get install cmake" before drivers install.

**Comment by danielcshn** [comment]
Hi, try this repository: https://github.com/danielcshn/LCD-show-kali Update to support Kali Linux Version 2019 or higher (Solution to Kernel Panic).

**Comment by GNexYde** [comment]
The solution is (READ ALL BEFORE DOING ANYTHING):
1) Copying the /boot/config.txt file from the SD card after flashing it with the Kali image, then put the card into Pi and boot, follow the steps to install on the LCD-show-kali repo.
2) The Kernel Panic will appear.
3) Power Pi down, pull card, reconnect to computer used for flashing (where the kali config.txt file is)
open the config.txt file on the card and the one saved earlier
4) On the version on the card, there is only about 5 or 6 lines you need to keep depending on the type of screen you are trying to enable. These should be the last lines in the file, remove everything before that and paste the content of your previously saved config.txt above those few lines.
5) save, eject, pull, insert, boot.....
 IF ALL WORKS JUST STOP AND ENJOY
If after booting the usb on the pi doesn't work you'll have to do some changes:
1) Before installing the lcd show stuff go into network management and set the wifi network you are using available for every user (so you'll have wifi before login and doesn't need a keyboard)
2) SSH into the pi and install xserver-xorg-input-libinput (sudo apt-get install xserver-xorg-input-libinput)
3) sudo reboot and enjoy the screen

This is the only method that worked for me (rpi4 4gb - kali 2020.3a arm64)

The original guide is from @n4kack, i just added the usb fix and shortened some parts

**Comment by danielcshn** [comment]
The solution is you do things well. Try this repository: https://github.com/danielcshn/LCD-show-kali

**Comment by pycvalade** [comment]
> This is the only method that worked for me (rpi4 4gb - kali 2020.3a arm64)

This worked for me on kali 2021.1-rpi4-64 with the 5 inch screen kit from Amazon:

1. Burn disk image from Kali website to sd card (I used balenaEtcher)
2. Remove and re-mount SD Card, copy config.txt file and create a blank file named "ssh" in there
3. Eject SD card and put it in Rpi4
4. Before booting plug ethernet cable and run nmap to see current network devices
5. wait 5 minutes
6. run nmap again, find the one for the raspberry pi and ssh into pi
7. sudo apt install xserver-xorg-input-libinput
8. run the LCD-show-kali script
9. when it asks for reboot, hit ctrl+c
10. sudo poweroff
8. once the ACT light (green) stop flashing, remove power
9. plug SD card back into computer
10. open the config.txt file and replace everything above the last section and replace with the whole original config.txt file you copied earlier (step 2)
11. eject SD card, put back in pi and power it up
12. Screen should work along with touch (worked for me at least)
